packages/legend-graph/src/models/protocols/pure/v1/transformation/pureGraph/to/helpers/V1_ValueSpecificationBuilderHelper.ts

```diff
@@ -869,6 +872,16 @@ export function V1_processProperty(
   if (inferredVariable instanceof AbstractPropertyExpression) {
     inferredType = inferredVariable.func.genericType.value.rawType;
   }
+  if (
+    inferredVariable instanceof SimpleFunctionExpression &&
+    matchFunctionName(
+      inferredVariable.functionName,
+      'meta::pure::functions::lang::subType',
+    )
+  ) {
+    inferredType =
+      inferredVariable.parametersValues[1]?.genericType?.value.rawType;
+  }
```

Comment:
Roundtrip Lamba Test ?

Reply:
As we discussed offline, I tested locally to make sure that added tests do go through the processors by writing some console.log statements.
(heads up: even though there is a typo or only tests themself without any necessary changes, tests still pass)